packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -942,12 +944,15 @@ export class ApolloServerBase<
    * `{req: express.Request, res: express.Response }` object) and to keep it
    * updated as you upgrade Apollo Server.
    */
-  public async executeOperation(
+  public async executeOperation<
+    TData = Record<string, any>,
+    TVariables = VariableValues,
+  >(
     request: Omit<GraphQLRequest, 'query'> & {
       query?: string | DocumentNode;
-    },
+    } & { variables?: TVariables | VariableValues },
```

Comment:
Do we want to add `variables` to the `Omit` above like `query` is?

I don't really understand what the `|` is doing here: why not just TVariables?